lib/hooks/useAuthorizationRolesMutation/useAuthorizationRolesMutation.js

```diff
@@ -0,0 +1,89 @@
+import noop from 'lodash/noop';
+import {
+  useMutation,
+  useQueryClient,
+} from 'react-query';
+
+import {
+  useNamespace,
+  useOkapiKy,
+} from '@folio/stripes/core';
+
+import { CAPABILITIES_LIMIT, ROLES_API } from '../../constants';
+
+const DEFAULT_VALUE = [];
+
+export const useAuthorizationRolesMutation = (options = {}) => {
+  const { tenantId, onSuccess = noop, ...otherOptions } = options;
+
+  const ky = useOkapiKy({ tenant: tenantId });
+  const queryClient = useQueryClient();
+  const [namespace] = useNamespace();
+
+  const { mutateAsync, isLoading } = useMutation({
+    mutationFn: async (roleId) => {
+      const capabilitiesSearchParams = {
+        limit: CAPABILITIES_LIMIT,
+        query: 'cql.allRecords=1 sortby resource',
+        expand: true
+      };
+      // fetch current role and its capabilities, capabilitySets
+      const currentRole = await ky
```

Comment:
Looks like these requests don't depend on each other. Let's use `await Promise.all([...])` in here to run the requests in parallel

Reply:
Updated